substrate/frame/treasury/src/migration.rs

```diff
@@ -0,0 +1,74 @@
+use super::*;
+use alloc::collections::BTreeSet;
+use core::marker::PhantomData;
+use frame_support::traits::UncheckedOnRuntimeUpgrade;
+
+/// The log target for this pallet.
+const LOG_TARGET: &str = "runtime::treasury";
+
+mod v1 {
+	use super::*;
+	pub struct MigrateToV1Impl<T, I>(PhantomData<(T, I)>);
+
+	impl<T: Config<I>, I: 'static> UncheckedOnRuntimeUpgrade for MigrateToV1Impl<T, I> {
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
+			let mut approval_index = BTreeSet::new();
+			for approval in Approvals::<T, I>::get().iter() {
+				approval_index.insert(*approval);
+			}
+
+			let mut proposals_released = 0;
+			for (proposal_index, p) in Proposals::<T, I>::iter() {
+				if !approval_index.contains(&proposal_index) {
+					let err_amount = T::Currency::unreserve(&p.proposer, p.bond);
+					debug_assert!(err_amount.is_zero());
```

Comment:
may be defensive? we do not have test for it, `debug_assert` wont really panic

Reply:
Added worst case scenario if amount is non zero to modify the proposal to reflect on bonds still stuck